File: frontend/src/app/dashboard/model/dto/dashboard.dto.ts
import { BaseChartOptionsDto } from './chart-options/base-chart-options.dto';
import { CompanyId, DashboardId, UserId } from '../../../shared/model/id-types';

/**
 * Used to deliver data from Backend to Frontend.
 * To transport data from Frontend to Backend see DashboardCreationDto.
 *
 * @see DashboardCreationDto
 */
export class DashboardDto {
  id: DashboardId;
  title: string;
  companyId: CompanyId;
  creatorId?: UserId;
  chartDtos: BaseChartOptionsDto[];
  creationDate: Date;
}

File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/dashboard/DashboardDto.java
package org.burningokr.dto.dashboard;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Collection;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardDto {
  private Long id;
  private String title;
  private Long companyId;
  private UUID creatorId;
  private Collection<BaseChartOptionsDto> chartDtos;
}
